rfcs/convergence/move-out-n-rename-makestyles.md

```diff
@@ -0,0 +1,94 @@
+
+### `makeStyles` is coupled to Fluent UI
+
+Even if there is interest to use `makeStyles` separately from Fluent UI, it's currently impossible since it's coupled with Fluent UI React v9. There is a separate proposal to decouple `makeStyles` from Fluent (see RFC [microsoft/fluentui#20651](https://github.com/microsoft/fluentui/pull/20651)).
+
+But while it stays as a folder in Fluent UI repo there is no clarity about how to use it for third parties.
+
+### Lack of docs
+
+Compared to other CSS-in-JS `makeStyles` does not have enough docs to explain APIs and design. While every other library has them 🙄 Currently there is no right place for them.
+
+### "Bad" naming
+
+It's a point that was not considered initially, but `makeStyles` is also a function in [MUI](https://mui.com/styles/basics/) and any search queries will point to it.
+
+### Opportunities for partners
+
+Without a strict dependency on Fluent UI v9 we can potentially involve more partners to be consumers of `makeStyles` since they will be able to try the new styling engine even without consuming v9 components.
+
+Partners using other component libraries can still see a performance improvement when using `makeStyles` with their components. This has been proven in the Teams chat list in T2.
+
+### Build issues
+
+Currently there are two unsolved issues: "buildless builds", "build dependency hell" once v9 & other packages are linked and circular dependencies with other repositories.
+
+#### Buildless environment
+
+The problem is described in [microsoft/fluentui#18357](https://github.com/microsoft/fluentui/issues/18357):
+
+- we cannot dogfood our Babel plugin/Webpack loader as it breaks buildless environment: to start Storybook you will need to run `yarn build --to @fluentui/make-styles-webpack-loader`
+- even currently it's required to build `@fluentui/jest-serializer-make-styles` before running tests
+
+#### Dependency issues
+
+For incremental adoption with Teams we are looking into a compat layer between `makeStyles` & Fela ([microsoft/fluentui#20611](https://github.com/microsoft/fluentui/pull/20611)). This requires Fluent UI React Northstar (v0) take `makeStyles` core as a dependency.
+
+Existing issues:
+
+- missing aliases for v9 packages in v0 configs
+- different TypeScript configuration
+- breaks in releases due different release cadences (v0 uses Lerna & manual releases)
+
+#### Circular dependencies between repos
+
+Currently `@fluentui/react-icons` depends on `@fluentui/react-make-styles`. When we need to perform major bump of `@fluentui/react-make-styles`:
+
+- try to bump `@fluentui/react-make-styles` to v10
+- `@fluentui/react-icons` still uses v10 (double dependency)
+- cannot bump `@fluentui/react-make-styles` in `@fluentui/react-icons` as v10 is not released yet 💥
+
+## Detailed Design or Proposal
+
+- Create new repository in Microsoft org and move related packages
+- Rename packages to use new name and NPM scope (`makeStyles` will not use Fluent UI scope)
```

Comment:
Would you use the `@microsoft` scope or a new one? Per [this page](https://docs.opensource.microsoft.com/releasing/publish/npm/) we're apparently not supposed to add new Microsoft-owned scopes without approval.

Reply:
> Would you use the @microsoft scope or a new one? 

I plan to create a new one and ensure that all required people have access to it.